pkg/curatedpackages/packagecontrollerclient.go

```diff
@@ -119,34 +121,66 @@ func (pc *PackageControllerClient) EnableCuratedPackages(ctx context.Context) er
 		noProxy := fmt.Sprintf("proxy.NO_PROXY=%s", strings.Join(pc.noProxy, "\\,"))
 		values = append(values, httpProxy, httpsProxy, noProxy)
 	}
-	if pc.eksaSecretAccessKey == "" || pc.eksaAccessKeyID == "" {
+	if (pc.eksaSecretAccessKey == "" || pc.eksaAccessKeyID == "") && pc.registryMirror == nil {
 		values = append(values, "cronjob.suspend=true")
 	}
 
-	err := pc.chartInstaller.InstallChart(ctx, pc.chart.Name, ociURI, pc.chart.Tag(), pc.kubeConfig, "", values)
+	writer, err := filewriter.NewWriter(pc.clusterName)
+	defer writer.CleanUpTemp()
 	if err != nil {
 		return err
 	}
+	var valueFilePath string
+	if valueFilePath, err = pc.CreateHelmOverrideValuesYaml(writer); err != nil {
+		return err
+	}
 
-	// Customers are currently requesting to show a warning in case
-	// credentials are not provided for curated packages
-	if err = pc.CreateCredentials(ctx); err != nil {
-		logger.MarkWarning("  Unable to create credentials for curated packages: ", "warning", err)
+	if err := pc.chartInstaller.InstallChart(ctx, pc.chart.Name, ociURI, pc.chart.Tag(), pc.kubeConfig, "", valueFilePath, values); err != nil {
+		return err
 	}
 
 	return pc.waitForActiveBundle(ctx)
 }
 
-// CreateCredentials creates necessary credentials to enable the installation of curated packages.
-// In order to create the credentials, there are two necessary steps:
-//   - Creating a kubernetes secret
-//   - Creating a single run of the cron job to ensure the secret is consumed
-func (pc *PackageControllerClient) CreateCredentials(ctx context.Context) error {
-	if err := pc.ApplySecret(ctx); err != nil {
-		return errors.New("unable to detect AWS authentication credentials for curated packages. Skipping package installation. Follow documentation to set credentials after cluster creation completes")
+// CreateHelmOverrideValuesYaml creates a temp file to override certain values in package controller helm install.
+func (pc *PackageControllerClient) CreateHelmOverrideValuesYaml(writer filewriter.FileWriter) (string, error) {
```

Comment:
Does this need to be exported?

Reply:
By exporting it, it makes the testing of this core functionality easier.